### PR TITLE
Add application configuration that can be used to configure services

### PR DESCRIPTION
**Pull request by prd-fox**
Add an initial version of configuration and its holder classes that can be sent around the application.

### CODE REVIEW

**Review by namtruong — DISMISSED** [review]

**Review by namtruong — APPROVED** [review]

File: nexus-app/src/main/java/com/github/nexus/config/ConfigHolder.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package com.github.nexus.config;
+
+import java.util.Objects;
+
+public enum ConfigHolder {
+
+    INSTANCE;
```
Comment (SatpalSandhu61) [inline nexus-app/src/main/java/com/github/nexus/config/ConfigHolder.java:7]: Could we not use longhand here, to make it clearer?
public static final MySingleton INSTANCE = new MySingleton();

Comment (prd-fox) [reply nexus-app/src/main/java/com/github/nexus/config/ConfigHolder.java:7]: Yeah, this was done to keep it outside the spring xml file, but then it somehow made its way in anyway. Definitely a refactoring consideration, but for now this works.

Comment (melowe) [reply nexus-app/src/main/java/com/github/nexus/config/ConfigHolder.java:7]: Using enum classes for singletons is pretty uncontroversial.

Comment (SatpalSandhu61) [reply nexus-app/src/main/java/com/github/nexus/config/ConfigHolder.java:7]: @Mark, sure but it's good to keep things simple. I'd never seen this before and I'm sure there are plenty of folks who'd have to do a quick google to work it out.

**Pull request merged**